app/models/conference.rb

```diff
@@ -81,6 +82,21 @@ def user_registered? user
   end
 
   ##
+  # Delete all EventSchedules that are not in the hours range
+  # After the conference has been successfully updated
+  def delete_event_schedules
+    if self.start_hour_changed? || self.end_hour_changed?
+      event_schedules = EventSchedule.select do |e|
```

Comment:
@Ana06 Okay so by `has_many thought` here you meant `has_many through`.I wasn't getting  what **thought** meant here and  was thinking that you might mean has many through association but then thought of confirming it first. But don't we generally use has many through associations  for many to many relation. Why do you want to use it for one to many relation between program and event schedule? Please correct me if am getting it all wrong.